node/network/availability-distribution/src/lib.rs

```diff
@@ -364,24 +375,22 @@ where
 {
 	let _timer = metrics.time_handle_our_view_change();
 
-	let old_view = std::mem::replace(&mut (state.view), view);
+	let old_view = std::mem::replace(&mut state.view, view);
 
 	// needed due to borrow rules
 	let view = state.view.clone();
-	let added = view.difference(&old_view).collect::<Vec<&'_ Hash>>();
 
 	// add all the relay parents and fill the cache
-	for added in added.iter() {
-		let added = **added;
-		let validators = query_validators(ctx, added).await?;
+	for added in view.difference(&old_view) {
+		let validators = query_validators(ctx, *added).await?;
 		let validator_index = obtain_our_validator_index(&validators, keystore.clone()).await;
 		state
-			.add_relay_parent(ctx, added, validators, validator_index)
+			.add_relay_parent(ctx, *added, validators, validator_index)
 			.await?;
 	}
 
 	// handle all candidates
-	for (candidate_hash, _receipt) in state.cached_live_candidates_unioned(added) {
+	for (candidate_hash, _receipt) in state.cached_live_candidates_unioned(view.difference(&old_view)) {
```

Comment:
So, previously we'd cached the result of `view.difference(&old_view)` as `added`. This PR removes that, and calculates the difference at least twice. `view.difference` requires `O(mn)` on the lengths of `view` and `old_view`. The implication of this change is that those sizes are small enough that recalculation is cheaper than allocation.

Does there exist profiling data to support that assertion?

Reply:
In like 99% of the time there is `1` element in `view`.